From f988b09f4382256e0288d85061a2765b27acf200 Mon Sep 17 00:00:00 2001
From: David Vernet <void@manifault.com>
Date: Thu, 4 Jan 2024 22:14:09 -0600
Subject: [PATCH 070/109] scx: Move sched_ext_entity.ddsq_id out of modifiable
 fields

When we added support for dispatching from ops.select_cpu(), I
accidentally put the sched_ext_entity.ddsq_id field into the "modifiable
fields" part of struct sched_ext_entity. It should be harmless, but
there shouldn't be any reason for a scheduler to muck with it, so let's
move it up.

Signed-off-by: David Vernet <void@manifault.com>
---
 include/linux/sched/ext.h | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/include/linux/sched/ext.h b/include/linux/sched/ext.h
index 73bcb9292..e5e69587b 100644
--- a/include/linux/sched/ext.h
+++ b/include/linux/sched/ext.h
@@ -693,6 +693,7 @@ struct sched_ext_entity {
 #ifdef CONFIG_SCHED_CORE
 	u64			core_sched_at;	/* see scx_prio_less() */
 #endif
+	u64			ddsq_id;
 
 	/* BPF scheduler modifiable fields */
 
@@ -717,12 +718,6 @@ struct sched_ext_entity {
 	 */
 	u64			dsq_vtime;
 
-	/*
-	 * Used to track when a task has requested a direct dispatch from the
-	 * ops.select_cpu() path.
-	 */
-	u64			ddsq_id;
-
 	/*
 	 * If set, reject future sched_setscheduler(2) calls updating the policy
 	 * to %SCHED_EXT with -%EACCES.
-- 
2.43.0.232.ge79552d197

